src/main/java/de/medizininformatikinitiative/torch/util/CrtdlParser.java

```diff
@@ -0,0 +1,70 @@
+package de.medizininformatikinitiative.torch.util;
+
+import ca.uhn.fhir.context.FhirContext;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.medizininformatikinitiative.torch.model.crtdl.Crtdl;
+import de.medizininformatikinitiative.torch.model.crtdl.DecodedCRTDLContent;
+import org.hl7.fhir.r4.model.Base64BinaryType;
+import org.hl7.fhir.r4.model.Parameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class CrtdlParser {
+    private static final Logger logger = LoggerFactory.getLogger(CrtdlParser.class);
+    private final FhirContext fhirContext;
+    private final ObjectMapper objectMapper;
+
+    public CrtdlParser(FhirContext fhirContext, ObjectMapper objectMapper) {
+        this.fhirContext = fhirContext;
+        this.objectMapper = objectMapper;
+    }
+
+
+    private Crtdl parseCrtdlContent(byte[] content) throws IOException {
+        return objectMapper.readValue(content, Crtdl.class);
+    }
+
+    public Mono<DecodedCRTDLContent> parseCrtdl(String body) {
```

Comment:
Mono is not needed here. Just throw an IAE in any case. Mapper functions can throw runtime exceptions. They are converted into errors automatically.